src/dsp_tools/commands/excel2json/resources.py

```diff
@@ -216,8 +221,22 @@ def excel2resources(
     # transform every row into a resource
     resources = [_row2resource(row, excelfile) for i, row in all_classes_df.iterrows()]
 
+    # check if resource names are unique
+    all_names = [r["name"] for r in resources]
+    if duplicates := {
+        index + 2: resdef["name"] for index, resdef in enumerate(resources) if all_names.count(resdef["name"]) > 1
+    }:
+        err_msg = (
+            f"Resource names must be unique inside every ontology, "
+            f"but your Excel file '{excelfile}' contains duplicates:\n"
+        )
+        for row_no, resname in duplicates.items():
+            err_msg += f" - Row {row_no}: {resname}\n"
```

Comment:
how about moving this logic into a util function? this seems too low level for such a high level function

Reply:
It is absolutely only parked there. This was in the json validation code, which is not actually what it is doing. When I do the fast fail (as mentioned above) this will get checked before anything is being constructed. So in the next PR it will be gone.